shared/chat/inbox/container/remote.js

```diff
@@ -13,10 +14,45 @@ const getMetaMap = (state: Container.TypedState) => [state.chat2.metaMap, state]
 const createShallowEqualSelector = Container.createSelectorCreator(Container.defaultMemoize, shallowEqual)
 
 // Get conversations
-const getMetas = Container.createSelector([getMetaMap], ([metaMap, state]) => [
-  metaMap.filter((meta, id) => Constants.isValidConversationIDKey(id)),
-  state,
-])
+const getMetas = Container.createSelector([getMetaMap], ([metaMap, state]) => {
+  const _username = state.config.username
+  /*
+    sort priority is determined by the following criteria:
+    - adhoc (DM's) chats come first
+    - mentioned channels
+    - recent activity in channel
+  */
+
+  const prioritizedMetas = metaMap.sort((a, b) => {
+    if (a.teamType === 'adhoc' && b.teamType === 'adhoc') {
+      // chats first
+      if (
+        Constants.getHasUnread(state, b.conversationIDKey) ||
+        Constants.getHasUnread(state, a.conversationIDKey)
+      ) {
+        // sort unread message by time
+        return b.timestamp - a.timestamp
+      }
```

Comment:
Could you explain the logic here? It seems even if we don't hit the condition, we'd end up at the [return after the conditions](https://github.com/keybase/client/pull/13829/files#diff-2e93add81efa99eb09c839d315c22ae9R51) below.

Perhaps you meant to check if only one of them has unread messages, in which case we'd put the unread at the beginning?

Reply:
It's working progress, I'm still trying to figure out how to get it right, I raised the PR so that I can get your assitance